cleanlab/dataset.py

```diff
@@ -0,0 +1,416 @@
+        "Overall Label Issues", "Inverse Label Noise Score", "Overall Label Quality Score" and a
+        description of each below. Noise scores are between 0 and 1, where 0 implies no label issues
+        in the class. The "Overall Label Quality Score" is also between 0 and 1 where 1 implies
+        perfect quality.
+        Columns:
+        * "Class Index"
+            The index of the class in 0, 1, ..., num_classes - 1.
+        * Overall Label Issues
+            count(given_label = k, true_label != k)
+            Estimated number of label issues in the class (usually the most accurate method).
+        * Inverse Label Issues
+            count(given_label != k, true_label = k)
+            Estimated number of times this class occurs in other classes in the dataset.
+        * Overall Label Noise Score
+            prob(true_label != k | given_label = k)
+            Estimated proportion of label issues in the class.
+        * Inverse Label Noise Score
+            prob(given_label != k | true_label = k)
+            Estimated proportion of times this class label occurs in other classes in the dataset.
+        * Overall Label Quality Score
+            p(given_label = k | true_label = k)
+            How accurate the labels are in each class as a conditional prob p(given=k | true=k).
+        By default, the DataFrame is ordered by "Overall Label Noise Score" descending.
+    """
+
+    if joint is None:
+        joint = estimate_joint(
+            labels=labels,
+            pred_probs=pred_probs,
+            confident_joint=confident_joint,
+            multi_label=multi_label,
+        )
+    if num_examples is None:
+        num_examples = _get_num_examples(labels=labels)
+    given_label_noise = joint.sum(axis=1) - joint.diagonal()  # p(s=k) - p(s=k,y=k) = p(y!=k, s=k)
+    true_label_noise = joint.sum(axis=0) - joint.diagonal()  # p(y=k) - p(s=k,y=k) = p(s!=k,y=k)
+    given_conditional_noise = given_label_noise / joint.sum(axis=1)  # p(y!=k, s=k) / p(s=k)
+    true_conditional_noise = true_label_noise / joint.sum(axis=0)  # p(s!=k, y=k) / p(y=k)
+    df = pd.DataFrame(
+        {
+            "Class Index": np.arange(len(joint)),
+            "Overall Label Issues": (given_label_noise * num_examples).round().astype(int),
+            "Inverse Label Issues": (true_label_noise * num_examples).round().astype(int),
+            "Overall Label Noise Score": given_conditional_noise,  # p(y!=k | s=k)
+            "Inverse Label Noise Score": true_conditional_noise,  # p(s!=k | y=k)
+            # Below could equivalently be computed as: joint.diagonal() / joint.sum(axis=1)
+            "Overall Label Quality Score": 1 - given_conditional_noise,  # p(y=k | s=k)
+        }
+    )
+    if class_names is not None:
+        df.insert(loc=0, column="Class Name", value=class_names)
+    return df.sort_values(by="Overall Label Noise Score", ascending=False)
```

Comment:
[Nit] future development will be less bug-prone if you define global variable:

OVERALL_LABEL_NOISE_SCORE = "Overall Label Noise Score"
and have all code refer to OVERALL_LABEL_NOISE_SCORE instead throughout.  This would ideally be done for all strings referenced in multiple places of the code.

Reply:
Great idea, will do later